packages/rs-drive-abci/src/validation/state_transition/identity_top_up.rs

```diff
@@ -62,68 +65,64 @@ impl StateTransitionValidation for IdentityTopUpTransition {
         };
 
         validation_result.set_data(Some(partial_identity));
+
+        let signature_validation =
+            validate_state_transition_signature_with_asset_lock_proof(self, platform)?;
+
+        if !signature_validation.is_valid() {
+            validation_result.merge(signature_validation);
+
+            return Ok(validation_result);
+        }
+
+        context.asset_lock_output = Some(signature_validation.into_data()?);
+
         Ok(validation_result)
     }
 
     fn validate_state<C: CoreRPCLike>(
         &self,
         platform: &PlatformRef<C>,
+        context: &mut ValidationDataShareContext,
         tx: TransactionArg,
     ) -> Result<ConsensusValidationResult<StateTransitionAction>, Error> {
-        let outpoint = match self.asset_lock_proof.out_point() {
-            None => {
-                return Ok(ConsensusValidationResult::new_with_error(
-                    ConsensusError::BasicError(
-                        BasicError::IdentityAssetLockTransactionOutputNotFoundError(
-                            IdentityAssetLockTransactionOutputNotFoundError::new(
-                                self.asset_lock_proof.instant_lock_output_index().unwrap(),
-                            ),
-                        ),
-                    ),
-                ));
-            }
-            Some(outpoint) => outpoint,
-        };
+        let mut validation_result = ConsensusValidationResult::default();
 
-        // Now we should check that we aren't using an asset lock again
-        let asset_lock_already_found = platform
-            .drive
-            .has_asset_lock_outpoint(&Bytes36(outpoint), tx)?;
-
-        if asset_lock_already_found {
-            let outpoint = OutPoint::from(outpoint);
-            return Ok(ConsensusValidationResult::new_with_error(
-                ConsensusError::BasicError(
-                    BasicError::IdentityAssetLockTransactionOutPointAlreadyExistsError(
-                        IdentityAssetLockTransactionOutPointAlreadyExistsError::new(
-                            outpoint.txid,
-                            outpoint.vout as usize,
-                        ),
-                    ),
-                ),
-            ));
+        let asset_lock_validation_result =
+            validate_asset_lock_proof_state(&self.asset_lock_proof, platform, tx)?;
+
+        if !asset_lock_validation_result.is_valid() {
+            validation_result.merge(asset_lock_validation_result);
+
+            return Ok(validation_result);
         }
 
-        self.transform_into_action(platform, tx)
+        self.transform_into_action(platform, context, tx)
```

Comment:
The order had an importance, we were validating against drive before we were asking from something from core. You just removed this order. Why?

Reply:
Not sure what do you mean about the order. Cheaper checks must be before and heave after to save resources and  people money. There is another thing wrong here. I forgot to move asset lock spent verification into chain proof validate state function